custom_components/pyscript/function.py

```diff
@@ -163,8 +163,16 @@ async def task_unique(name, kill_me=False):
                     except asyncio.CancelledError:
                         pass
             if curr_task in cls.our_tasks:
+                if name in cls.unique_name2task:
+                    task = cls.unique_name2task[name]
+                    if task in cls.unique_task2name:
+                        if name in cls.unique_task2name[task]:
+                            cls.unique_task2name[task].remove(name)
```

Comment:
It would be more efficient to use a set, in which case the last two lines can just be a single-line `discard` (which is a no-op if not present)